props-core/src/main/java/sh/props/Registry.java

```diff
@@ -53,10 +52,12 @@
   final List<Layer> layers = new ArrayList<>();
   final ConcurrentHashMap<String, HashSet<AbstractProp<?>>> notifications =
       new ConcurrentHashMap<>();
+  private final Scheduler scheduler;
 
   /** Ensures a registry can only be constructed through a builder. */
-  Registry() {
+  Registry(Scheduler scheduler) {
```

Comment:
bind a scheduler to the registry, to allow configuring the executor/fork-join pool sizes/strategies.